### DISCUSSION TITLE
Support more programming languages in code block and intelligent language recognition.

### DISCUSSION THREAD

**Discussion by dkkb** [outline/outline#2787] [Ideas]
1.Until now, Outline hasn't supported rust-lang, which would be a fantastic addition if we could support it.

<img src="https://user-images.githubusercontent.com/82504881/142595184-6826bdb1-1364-4a67-8826-835d7804ca63.png" width=200>

2.If the editor can recognize the language automatically when we copy code into the code block, that would be fantastic.

3.We could also add more themes to the code block.

e.g.
<img src="https://user-images.githubusercontent.com/82504881/142595969-97db44c1-9fd4-4a4a-b775-e4ee7dd9473d.png">

**Comment by tommoor** [comment]
Rust and Objective-C support will be in the next release. Generally we've tried to keep this minimal as every additional language and feature slows down the app and increases loading times. The code block is particularly sensitive to this as additional language variations add weight but are only used by a very small percentage of users
